### PR TITLE
Fix: drop table statement

### LINKED ISSUE #1982: table("public.my_table").drop() does not work
Dropping a table by calling the following will fail when using PostgreSQL:

```
const instance = spanner.instance('my-instance');
const database = instance.database('my-database');
const table = database.table('public.Singers');
table.delete();
```

The reason is that [this line](https://github.com/googleapis/nodejs-spanner/blob/b0df047cf57d1a14e6c29e891ea54f96c2e1ef97/src/table.ts#L346) fails to take two things into consideration:
1. PostgreSQL uses " (double quotes) to quote identifiers, and not ` (backtick).
2. If the table name is prefixed with a schema name, then the entire string is quoted, instead of the schema name and table name separately.

The output is:
- Current: `` DROP TABLE `public.Singers` ``
- Expected: `DROP TABLE "public"."Singers"`

### PR DESCRIPTION
**Pull request by alkatrivedi**
Fixes: #1982

### CODE REVIEW

File: src/database.ts [hunk 1]
```diff
@@ -120,6 +122,15 @@ type CreateBatchTransactionResponse = [
   google.spanner.v1.ITransaction | google.spanner.v1.ISession,
 ];
 type DatabaseResponse = [Database, r.Response];
+export interface DialectCallback {
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/database.ts]: Removed

File: src/database.ts [hunk 2]
```diff
@@ -1476,6 +1504,57 @@ class Database extends common.GrpcServiceObject {
     return metadata.state || undefined;
   }
 
+  /**
+   * Get this database's dialect.
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/database.ts]: Updated

File: src/database.ts [hunk 3]
```diff
@@ -1476,6 +1504,57 @@ class Database extends common.GrpcServiceObject {
...
+    optionsOrCallback?: CallOptions | GetDatabaseDialectCallback,
+    cb?: GetDatabaseDialectCallback
+  ): Promise<
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.DatabaseDialect>
+    | undefined
+  > {
+    const gaxOptions =
+      typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
+
+    const callback =
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/database.ts]: Updated

File: src/table.ts [hunk 4]
```diff
@@ -342,11 +342,37 @@ class Table {
...
-    return this.database.updateSchema(
-      'DROP TABLE `' + this.name + '`',
-      gaxOptions,
-      callback!
-    );
+    const parts = this.name.split('.');
+    const schema = parts.length === 2 ? parts[0] : null;
+    const table = parts.length === 2 ? parts[1] : this.name;
+
+    let dropStatement = 'DROP TABLE `' + this.name + '`';
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/table.ts]: Updated

File: src/table.ts [hunk 5]
```diff
@@ -342,11 +342,37 @@ class Table {
     const callback =
       typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
 
-    return this.database.updateSchema(
-      'DROP TABLE `' + this.name + '`',
-      gaxOptions,
-      callback!
-    );
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/table.ts:349]: Done

File: src/database.ts [hunk 6]
```diff
@@ -1476,6 +1504,57 @@ class Database extends common.GrpcServiceObject {
...
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
+
+    if (
```
[the remark this answers is not in the document]
Comment (alkatrivedi) [reply src/database.ts:1531]: Lint is failing on this change, hence skipping the change

File: src/database.ts [hunk 7]
```diff
@@ -1476,6 +1491,54 @@ class Database extends common.GrpcServiceObject {
     return metadata.state || undefined;
   }
 
+  /**
+   * Get this database's dialect to retrieves the dialect of the database
```
Comment (surbhigarg92) [inline src/database.ts:1495]:
```suggestion
   * Retrieves the dialect of the database
```

File: src/database.ts [hunk 8]
```diff
@@ -1476,6 +1491,54 @@ class Database extends common.GrpcServiceObject {
...
+   *
+   * @param {object} [gaxOptions] Request configuration options,
+   *     See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions}
+   *     for more details.
+   * @param {GetDatabaseDialectCallback} [callback] Callback function.
+   * @returns {Promise<EnumKey<typeof, databaseAdmin.spanner.admin.database.v1.DatabaseDialect> | undefined>}
+   *     When resolved, contains the database dialect of the database if the dialect
+   *     is defined.
+   */
+
```
Comment (surbhigarg92) [inline src/database.ts:1509]: Add an example in the documentation like we have in get state method



   ```
   @example
   * const {Spanner} = require('@google-cloud/spanner');
   * const spanner = new Spanner();
   * const instance = spanner.instance('my-instance');
   * const database = instance.database('my-database');
   * const dialect = await database.getDialect();
   * const isGoogleSQL = (dialect === 'GOOGLE_STANDARD_SQL');
   ```

File: src/table.ts [hunk 9]
```diff
@@ -342,11 +342,37 @@ class Table {
...
+    const [schema, table] = this.name.includes('.')
+      ? this.name.split('.')
+      : [null, this.name];
+
+    let dropStatement = 'DROP TABLE `' + table + '`';
+
+    const performDelete = async (): Promise<void | DropTableResponse> => {
+      const result = await this.database.getDatabaseDialect(gaxOptions);
+
+      if (result && result.includes('POSTGRESQL')) {
```
Comment (surbhigarg92) [inline src/table.ts:354]: nit: Define constants and use that for `GOOGLE_STANDARD_SQL` and `POSTGRESQL`

**Review by surbhigarg92 — APPROVED** [review]

**Comment by surbhigarg92** [comment]
@olavloite Could also please review this PR once.

File: src/database.ts (hunk 7 above)
Comment (alkatrivedi) [reply src/database.ts]: Done

File: src/database.ts (hunk 8 above)
Comment (alkatrivedi) [reply src/database.ts:1517]: Done

**Review by olavloite — APPROVED** [review]

**Pull request merged**